src/hardware/serial_terminal.rs

```diff
@@ -25,28 +85,128 @@ impl Write for OutputBuffer {
     }
 }
 
+impl core::fmt::Write for Context {
+    /// Write data to the serial terminal.
+    ///
+    /// # Note
+    /// The terminal uses an internal buffer. Overflows of the output buffer are silently ignored.
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        self.output.write_str(s)
+    }
+}
+
+fn handle_list(
+    _menu: &menu::Menu<Context>,
+    _item: &menu::Item<Context>,
+    _args: &[&str],
+    context: &mut Context,
+) {
+    writeln!(context, "Available properties:").unwrap();
```

Comment:
Can this also print the docstring (or some other struct-associated string) of the settings struct?

Reply:
Right now, no. I think we could expand the miniconf macro to expose the docstrings as metadata as well, but that would be a modification of miniconf. I'll open an issue.

Reply:
I added https://github.com/quartiq/miniconf/issues/187 for this functionality

Reply:
I meant just the docstring of the struct, not of the fields. That would already cover a lot.